Consul/Client_Results.cs

```diff
@@ -44,6 +45,11 @@ public ConsulResult(ConsulResult other)
     /// </summary>
     public class QueryResult : ConsulResult
     {
+        /// <summary>
+        /// The headers returned by the Consul server
+        /// </summary>
+        public HttpResponseHeaders Headers { get; set; }
```

Comment:
I'm not sure about exposing the  `HttpResponseHeaders` in the response object. It changes the contract of this class dramatically, and the `HttpResponseHeaders` class is very implementation specific. I think it is better to use simpler types and add two extra fields for 'X-Cache" and 'Age' headers.

Reply:
I agree that it gives the user much flexibility. So I understand your motivation, but I believe that exposing the `HttpResponseHeaders` would be a bad API design practice. 
It is an implementation-specific class. Once we would expose it we would never be able to change our implementation. I'm not saying that we are ever going to do that, but I'm saying that it is going to strongly couple implementation and the API together which is a bad practice. 
The number of different HTTP headers that are going to be used by the consul is very limited, so we can parse them in our library for the user. If the number of possible headers was very high, we would need more flexibility as you suggest.

Reply:
I see, that makes a lot of sense.

Okay, in that case, we can declare it as a private property and retrieve the value from our test using reflection?

Reply:
> I see, that makes a lot of sense.
> 
> Okay, in that case, we can declare it as a private property and retrieve the value from our test using reflection?

I don't think we need reflection, I think the best way to do this is by adding two new extra public properties (`XCache` and `Age`) in the `QueryResult class`.  Then update the `ParseQueryHeaders` method to parse those headers. It will make it possible to implement the test, but also it will make it possible for the end user to know whether the result comes from the cache.

Reply:
I am wondering if we need to update the following constructor overload for `QueryResult`.

```cs
public QueryResult(QueryResult other) : base(other)
{
    LastIndex = other.LastIndex;
    LastContact = other.LastContact;
    KnownLeader = other.KnownLeader;
}
```

to

```cs
public QueryResult(QueryResult other) : base(other)
{
    LastIndex = other.LastIndex;
    LastContact = other.LastContact;
    KnownLeader = other.KnownLeader;
   AddressTranslationEnabled = other.AddressTranslationEnabled;
   XCache = other.XCache;
  Age = Age;
}
```